### PR TITLE
docs(Scalar.AspNetCore): add .NET integration documentation

### LINKED ISSUE #3617: Add .NET instructions under documentation/integrations
### What happens?

Currently, the [documentation/integrations directory](https://github.com/scalar/scalar/tree/4cf8c529a48794b1f77140c888da347b2e1cb2da/documentation/integrations) doesn't contain docs for the .NET integration. We should copy them over here.

### What did you expect to happen?

I would expect to see a `dotnet.md` file with instructions on how to use Scalar's `Scalar.AspNetCore` NuGet package.

### OpenAPI Document

_No response_

### PR DESCRIPTION
**Pull request by xC0dex**
This PR addresses the missing .NET integration documentation in the `documentation/integrations` folder. I've added some useful examples and descriptions.

Closes #3617 

For easier reading: https://github.com/xC0dex/scalar/blob/docs/documentation/integrations/dotnet.md

**Note**: The `Lint Markdown files` flow is failing due to a dead reference. Once this is merged, the link is no longer dead 😆.

### CODE REVIEW

**Comment by xC0dex** [comment]
Hey @captainsafia,
You offered to review my PR, and I’d love to take you up on that 👀.
Thanks!

**Comment by hanspagel** [comment]
Wow, great work!

**Review by marclave — APPROVED** [review]
AWESOME WORK @xC0dex !!


@captainsafia just opened this issue, you added the docs so quick!

File: documentation/integrations/dotnet.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        {
+            oauth.ClientId = "your-client-id";
+            oauth.Scopes = ["profile"];
+        });
+});
+```
+
+### OpenAPI Document
+
+Scalar expects the OpenAPI document to be located at `/openapi/{documentName}.json`, matching the route of the built-in .NET OpenAPI generator. If the document is located elsewhere (e.g., when using Swashbuckle or NSwag), specify the path using the `OpenApiRoutePattern` property:
```
Comment (captainsafia) [inline documentation/integrations/dotnet.md:80]:
```suggestion
Scalar expects the OpenAPI document to be located at `/openapi/{documentName}.json`, matching the route of the built-in .NET OpenAPI generator in the `Microsoft.AspNetCore.OpenApi` package. If the document is located elsewhere (e.g., when using Swashbuckle or NSwag), specify the path using the `OpenApiRoutePattern` property:
```

**Review by captainsafia — COMMENTED** [review]
Looks great overall! I provided a few small comments inline.

**Comment by xC0dex** [comment]
Thanks for the feedback. I've applied your suggestions!

**Comment by xC0dex** [comment]
I wondered if it might make sense to reference this documentation in the [official .NET docs](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/openapi/using-openapi-documents?view=aspnetcore-9.0#use-scalar-for-interactive-api-documentation)?

**Comment by hanspagel** [comment]
Wow, what a great PR and the collaboration between you @captainsafia and @xC0dex 👐

**Review by hanspagel — APPROVED** [review]

**Pull request merged**